packages/launcher/src/index.tsx

```diff
@@ -322,6 +572,80 @@ export namespace ILauncher {
      */
     kernelIconUrl?: string;
   }
+
+  export interface IGroupedItemOptions {
```

Comment:
This interface ends up mixing two ideas: 1) grouping and 2) kernel specific logic. Those two concepts need to be separated, so there is a clear outer loop over groups, and then kernel specific logic for one of the sections (probably a separate `<KernelSection>` component that knows how to render the notebook/console cards).

Reply:
Also, within the `<KernelSection>` component there should be another groupby on the kernel name that will handle the notebook/console grouping.